e2e/testcases/ignore_mutation_test.go

```diff
@@ -0,0 +1,429 @@
+	nt.Must(rootSyncGitRepo.Remove("acme/ns.yaml"))
+	nt.Must(rootSyncGitRepo.CommitAndPush("prune namespace-bookstore"))
+	nt.Must(nt.WatchForAllSyncs())
+
+	// Namespace should have been pruned
+	nt.Must(nt.Watcher.WatchForNotFound(kinds.Namespace(), namespace.Name, ""))
+
+	nt.T.Log("Add back namespace to Git")
+	namespace = k8sobjects.NamespaceObject("bookstore",
+		core.Annotation("season", "winter"), core.Annotation(metadata.LifecycleMutationAnnotation, metadata.IgnoreMutation))
+	nt.Must(rootSyncGitRepo.Add("acme/ns.yaml", namespace))
+	nt.Must(rootSyncGitRepo.CommitAndPush("add a namespace"))
+
+	nt.Must(nt.Watcher.WatchObject(kinds.Namespace(), namespace.Name, "",
+		testwatcher.WatchPredicates(
+			testpredicates.HasAnnotation("season", "winter"),
+			testpredicates.HasAnnotation(metadata.LifecycleMutationAnnotation, metadata.IgnoreMutation),
+		)))
+}
+
+// TestAnnotationDrift verifies that the `season` annotation is correct when modified with Kubectl and
+// in the declared config
+func TestAnnotationDrift(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.DriftControl, ntopts.SyncWithGitSource(nomostest.DefaultRootSyncID, ntopts.Unstructured))
+	rootSyncGitRepo := nt.SyncSourceGitReadWriteRepository(nomostest.DefaultRootSyncID)
+
+	nt.T.Log("Adding namespace to Git")
+	namespace := k8sobjects.NamespaceObject("bookstore",
+		core.Annotation(metadata.LifecycleMutationAnnotation, metadata.IgnoreMutation))
+	nt.Must(rootSyncGitRepo.Add("acme/ns.yaml", namespace))
+	nt.Must(rootSyncGitRepo.CommitAndPush("add a namespace"))
+	nt.Must(nt.WatchForAllSyncs())
+
+	nt.T.Log("Add new annotation to the declared namespace")
+	updatedNamespace := k8sobjects.NamespaceObject("bookstore",
+		core.Annotation(metadata.LifecycleMutationAnnotation, metadata.IgnoreMutation),
+		core.Annotation("season", "summer"))
+	nt.Must(rootSyncGitRepo.Add("acme/ns.yaml", updatedNamespace))
+	nt.Must(rootSyncGitRepo.CommitAndPush("update namespace"))
+
+	nt.Must(nt.Watcher.WatchObject(kinds.Namespace(), "bookstore", "",
+		testwatcher.WatchPredicates(
+			testpredicates.MissingAnnotation("season"),
+			testpredicates.HasAnnotationKey(metadata.LifecycleMutationAnnotation))))
+
+	nt.T.Log("Add the 'season=winter' annotation to the namespace using kubectl")
+	nsObj := namespace.DeepCopy()
+	nsObj.Annotations["season"] = "winter"
+	nt.Must(nt.KubeClient.Apply(nsObj))
+
+	nt.Must(nt.Watcher.WatchObject(kinds.Namespace(), nsObj.Name, "",
+		testwatcher.WatchPredicates(
+			testpredicates.HasAnnotation("season", "winter"),
+		)))
+
+	// Push a new commit to trigger a new apply.
+	nt.T.Log("Add another namespace to Git to run the applier")
+	nsObj2 := k8sobjects.NamespaceObject("new-ns")
+	nt.Must(rootSyncGitRepo.Add("acme/ns2.yaml", nsObj2))
+	nt.Must(rootSyncGitRepo.CommitAndPush("add another namespace"))
+
+	nt.Must(nt.Watcher.WatchObject(kinds.Namespace(), nsObj.Name, "",
+		testwatcher.WatchPredicates(
+			testpredicates.HasAnnotation("season", "winter"),
```

Comment:
Besides validating the `season` annotation, we should also verify the value of the `configmanagement.gke.io/token` annotation on the Namespace is changed.

Without this, it is hard to tell if the Namespace is updated or not because Line240 already indicates it has a `season: winter` annotation.

Reply:
Done